### PR TITLE
Create docs/research and add @gabalafou's docs

### PR DESCRIPTION
**Pull request by gabalafou**

### CODE REVIEW

**Comment by gabalafou** [comment]
This is a first pass. I'm sure I didn't do this correctly, but I wanted to go ahead and open a PR to collaborate and learn-by-doing in the process

**Review by isabela-pf — APPROVED** [review]
Thanks for contributing this! 

I can't verify the accuracy of all the content, but naming and relationships and such seem correct to me. Everything looks well and properly linked, too. I really appreciate the consideration of including a date and version for this.

Headings and everything look correctly used, so proper HTML should come from this markdown. All images have relevant alt text. All links have good descriptions.

Overall, I can only imagine how much work this took. Great job! Endless thanks to you! Totally worth merging in my opinion.

**Note:** If there's no pushback against this in a week, I'll go ahead and merge it. We can always revisit.

File: docs/source/research/map-jupyterlab-frontend-architecture/README.md [hunk 1]
```diff
@@ -0,0 +1,1117 @@
+# Mapping JupyterLab's user interface to source code
+
+Author: Gabriel Fouasnon
+
+Date: January 18, 2022
```
Comment (trallard) [inline docs/source/research/map-jupyterlab-frontend-architecture/README.md:5]: It would make more sense to have this as `last updated`

**Review by trallard — COMMENTED** [review]
I finally read through *all* of this doc and not just skimmed - I feel this document adds a much-needed component view of Lab and it is really easy to follow  👏🏽 very well done!

File: docs/source/research/map-jupyterlab-frontend-architecture/README.md (hunk 1 above)
Comment (gabalafou) [reply docs/source/research/map-jupyterlab-frontend-architecture/README.md]: Updated!

**Comment by gabalafou** [comment]
@trallard thank you! it would be great if at some point this work helps a future frontend developer get oriented to the JupyterLab codebase

**Pull request merged**